DataStructures/Graphs/UndirectedWeightedDenseGraph.cs

```diff
@@ -109,10 +109,10 @@ public virtual IEnumerable<WeightedEdge<T>> Edges
         /// <summary>
         ///     Get all incoming edges to a vertex
         /// </summary>
-        public virtual IEnumerable<WeightedEdge<T>> IncomingEdges(T vertex)
+        public new virtual IEnumerable<WeightedEdge<T>> IncomingEdges(T vertex)
         {
             if (!HasVertex(vertex))
-                throw new ArgumentOutOfRangeException("One of vertex is not part of the graph.");
+                throw new ArgumentOutOfRangeException($"One of vertex is not part of the graph.");
```

Comment:
I think we shouldn't add a "$" before strings if we don't use string interpolation.
But it would be a good idea to use interpolation for printing the vertex.

Reply:
So I suggest to print the actual vertex here with interpolation.